src/agent.rs

```diff
@@ -434,6 +436,32 @@ impl AgentBuilder {
         self
     }
 
+    /// The user-agent header to associate with all requests from this agent by default.
+    ///
+    /// Defaults to `ureq/[VERSION]`. You can override the user-agent on an individual request by
+    /// setting the `User-Agent` header when constructing the request.
+    ///
+    /// ```
+    /// # fn main() -> Result<(), ureq::Error> {
+    /// # ureq::is_test(true);
+    /// let agent = ureq::builder()
+    ///     .user_agent("ferris/1.0".into())
+    ///     .build();
+    /// // Uses agent's header
+    /// let result = agent.get("http://httpbin.org/headers").call()?;
+    ///
+    /// // Overrides user-agent set on the agent
+    /// let result = agent.get("http://httpbin.org/headers")
+    ///     .set("User-Agent", "super-ferris/2.0")
+    ///     .call()?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn user_agent(mut self, user_agent: String) -> Self {
```

Comment:
I'd like this to be `user_agent: &str` to be analogous with the rest of the API, see `Request::set` for instance.

Reply:
SGTM